File: simple_parenthesis_removal_js/remove_parenthesis.js
function solve(s) {
  let res = ''
  let op = 1
  let stack = [1]

  for (const x in s) {
    if (s[x] === '(') {
      stack.push(op)
    } else if (s[x] === ')') {
      stack.pop()
    } else if (s[x] === '+') {
      op = stack[stack.length - 1]
    } else if (s[x] === '-') {
      op = -stack[stack.length - 1]
    } else {
      if (res || op !== 1) {
        res += op === 1 ? '+' : '-'
      }
      res += s[x]
    }
  }

  return res
}
module.exports = solve;

File: python/simple_parenthesis_removal/remove_parenthesis.py
def solve(s):
    res, op, stack = "", 1, [1]

    for x in s:
        if x == "(":
            stack.append(op)
        elif x == ")":
            stack.pop()
        elif x == "+":
            op = stack[-1]
        elif x == "-":
            op = -stack[-1]
        else:
            if res or op != 1:
                res += "+" if op == 1 else "-"
            res += x

    return res
